applications/GeoMechanicsApplication/custom_conditions/general_U_Pw_diff_order_condition.cpp

```diff
@@ -70,34 +71,9 @@ void GeneralUPwDiffOrderCondition::Initialize(const ProcessInfo& rCurrentProcess
 
 //----------------------------------------------------------------------------------------
 
-void GeneralUPwDiffOrderCondition::
-    GetDofList( DofsVectorType& rConditionDofList,
-                const ProcessInfo& rCurrentProcessInfo ) const
+void GeneralUPwDiffOrderCondition::GetDofList(DofsVectorType& rConditionDofList, const ProcessInfo&) const
 {
-    KRATOS_TRY
-
-    const GeometryType& rGeom = GetGeometry();
-    const SizeType Dim = rGeom.WorkingSpaceDimension();
-    const SizeType NumUNodes = rGeom.PointsNumber();
-    const SizeType NumPNodes = mpPressureGeometry->PointsNumber();
-    const SizeType ConditionSize = NumUNodes * Dim + NumPNodes;
-
-    if(rConditionDofList.size() != ConditionSize)
-        rConditionDofList.resize(ConditionSize);
-
-    SizeType Index = 0;
-
-    for(SizeType i = 0; i < NumUNodes; ++i) {
-        rConditionDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_X );
-        rConditionDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_Y );
-        if (Dim > 2)
-            rConditionDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_Z );
-    }
-
-    for (SizeType i=0; i<NumPNodes; ++i)
-        rConditionDofList[Index++] = GetGeometry()[i].pGetDof( WATER_PRESSURE );
-
-    KRATOS_CATCH( "" )
+    rConditionDofList = GetDofs();
```

Comment:
I wonder whether this should have an implementation or should it be abstract and always overridden by a specific implementation.

Reply:
As indicated before, we might be able to simplify things even more after we have rearranged the DOFs. But since we intend to replace these element classes (as part of the element redesign), we'd better do this right when we implement the new elements.